app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/WifiDirectManager.kt

```diff
@@ -243,8 +243,19 @@ class WifiDirectManager @Inject constructor(
     if (BuildConfig.DEBUG) {
       Log.d(TAG, "Starting handshake")
     }
-    peerGroupHandshakeAsyncTask = PeerGroupHandshakeAsyncTask(this)
-      .also { it.execute() }
+    peerGroupHandshake = PeerGroupHandshake(this)
+      .also {
+        lifecycleCoroutineScope.launch {
+          val inetAddress = it.handshake()
+          inetAddress?.let(::setClientAddress)
```

Comment:
I'd say avoid `let`ing and `elvis`ing, if/else will just read a lot cleaner

Reply:
done